x/language.rb

```diff
@@ -0,0 +1,15 @@
+module X
+  class Language
+    def self.of(id)
+      tracks.fetch(id.to_s) do
+        @@tracks[id.to_s] = Track.find(id).language
```

Comment:
Here is the simple caching I implemented. I'm not sure though what `X::Track.find` should return when the id isn't found. Maybe we should write some code for that.